src/cplscheme/BiCouplingScheme.hpp

```diff
@@ -84,6 +85,23 @@ class BiCouplingScheme : public BaseCouplingScheme {
     return _receiveData;
   }
 
+  /**
+   * @brief BiCouplingScheme has _sendData and _receiveData
+   * @returns DataMap with all data
+   */
+  DataMap getAllData() override
+  {
+    DataMap allData;
+    // @todo use C++17 std::map::merge
+    for (auto &pair : _sendData) {
+      allData.emplace(pair);
+    }
+    for (auto &pair : _receiveData) {
+      allData.emplace(pair);
+    }
+    return allData;
```

Comment:
You can simply copy A and add B

```suggestion
    DataMap allData{_sendData};
    allData.insert(_receiveData.begin(), _receiveData.end());
    return allData;
```

Reply:
Fixed in https://github.com/precice/precice/pull/1185/commits/854ae5fabe13b0709de5b1282301fa818d3af0c6.